gobblin-core-base/src/main/java/gobblin/writer/Batch.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package gobblin.writer;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gobblin.annotation.Alpha;
+
+
+/**
+ * An Batch class that contains all the records pushed by {@link BatchAccumulator}.
+ * Also it tracks each record's offset in the batch and wrapped it into {@link RecordMetaData}
+ * and return as a future object
+ */
+@Alpha
+public abstract class Batch<D>{
+
+  private final List<Thunk> thunks;
+  private final long creationTimestamp;
+  private final long id;
+  private long recordCount;
+  private final CountDownLatch latch = new CountDownLatch(1);
+  private static AtomicInteger identifier = new AtomicInteger(0);
+  private static final Logger LOG = LoggerFactory.getLogger(Batch.class);
+
+  public Batch () {
+    recordCount = 0;
+    thunks = new ArrayList<>();
+    creationTimestamp = System.currentTimeMillis();
+    id = identifier.incrementAndGet();
+  }
+
+  public void done() {
+    latch.countDown();
+  }
+
+  public boolean isTTLExpire() {
+    return (System.currentTimeMillis() - creationTimestamp) >= 3000;
```

Comment:
Also I moved this isTTLExpire to EventhubBatch because this is the implementation details